pkg/shp/streamer/tar.go

```diff
@@ -65,3 +66,27 @@ func NewTar(src string) (*Tar, error) {
 	t := &Tar{src: src}
 	return t, t.bootstrap()
 }
+
+func GetTarSize(src string) (*Tar, error) {
+	t := &Tar{src: src}
+	return t, t.tarSize()
+}
+
+func (t *Tar) tarSize() error {
+	var size int64
+	filepath.Walk(t.src, func(fpath string, stat fs.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if t.skipPath(fpath, stat) {
+			return nil
+		}
+		size += stat.Size()
+		header, err := tar.FileInfoHeader(stat, stat.Name())
+		header.Name = trimPrefix(t.src, fpath)
+		size += header.Size
+		return err
+	})
+	t.Size = size
+	return nil
+}
```

Comment:
I am not quite sure about the error handling here. Can you have a look at my suggestion and tell me what you think about it? Rational was that I do not think the errors of the `Walk` function should be ignored. Also, I think we should check the error of `FileInfoHeader` and react immediately.
```suggestion
func (t *Tar) tarSize() error {
	var size int64

	err := filepath.Walk(t.src, func(fpath string, stat fs.FileInfo, err error) error {
		if err != nil {
			return err
		}
		if t.skipPath(fpath, stat) {
			return nil
		}

		size += stat.Size()

		header, err := tar.FileInfoHeader(stat, stat.Name())
		if err != nil {
			return err
		}

		header.Name = trimPrefix(t.src, fpath)
		size += header.Size
		return nil
	})

	t.Size = size
	return err
}
```

Reply:
Or, if we want to switch to `WalkDir` instead of `Walk`. I think `Walk` is not deprecated, but the docs suggest to use `WalkDir` instead.
```suggestion
func (t *Tar) tarSize() error {
	var size int64

	err := filepath.WalkDir(t.src, func(fpath string, d fs.DirEntry, err error) error {
		if err != nil {
			return err
		}

		stat, err := d.Info()
		if err != nil {
			return err
		}

		if t.skipPath(fpath, stat) {
			return nil
		}

		size += stat.Size()

		header, err := tar.FileInfoHeader(stat, stat.Name())
		if err != nil {
			return err
		}

		header.Name = trimPrefix(t.src, fpath)
		size += header.Size
		return nil
	})

	t.Size = size
	return err
}
```